lib/Jsrt/Jsrt.cpp

```diff
@@ -742,24 +741,33 @@ CHAKRA_API JsSetCurrentContext(_In_ JsContextRef newContext)
 
 #if ENABLE_TTD
         Js::ScriptContext* newScriptContext = newContext != nullptr ? static_cast<JsrtContext*>(newContext)->GetScriptContext() : nullptr;
+        Js::ScriptContext* oldScriptContext = currentContext != nullptr ? static_cast<JsrtContext*>(currentContext)->GetScriptContext() : nullptr;
+
         if(newScriptContext == nullptr)
         {
-            Js::ScriptContext* oldScriptContext = currentContext != nullptr ? static_cast<JsrtContext*>(currentContext)->GetScriptContext() : nullptr;
-            if(currentContext != nullptr && oldScriptContext->IsTTDRecordModeEnabled())
+            if(oldScriptContext == nullptr)
+            {
+                ; //if newScriptContext and oldScriptContext are null then we don't worry about doing anything
+            }
+            else
             {
-                if(oldScriptContext->ShouldPerformRecordAction())
+                if(oldScriptContext->IsTTDRecordModeEnabled()) 
                 {
-                    oldScriptContext->GetThreadContext()->TTDLog->RecordJsRTSetCurrentContext(_actionEntryPopper, nullptr);
-                }
+                    //already know newScriptContext != oldScriptContext so don't check again
```

Comment:
When did we do this check?  I don't see anything that confirms this comment's assertion.